### PR TITLE
Feature/ivp 1156 initial drop

### PR DESCRIPTION
**Pull request by SteveHicksCH**
Team internal review 

Hopefully just some more POJO tests and a mockito test for the service

Then I will add some documentation to the LLD for use in a "formal" review

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/ocr/api/ThreadConfig.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        executor.setMaxPoolSize(threadPoolSize);
+        executor.setThreadNamePrefix("task_executor_thread");
+        executor.initialize();
+        return executor;
+    }
+
+    private int getThreadPoolSize() {
+
+        var configuredThreadPoolSize = System.getenv("OCR_TESSERACT_THREAD_POOL_SIZE");
+        int threadPoolSize = DEFAULT_THREAD_POOL_SIZE;
```
Comment (ljgeorge) [inline src/main/java/uk/gov/companieshouse/ocr/api/ThreadConfig.java:37]: I take it we will make threadPoolSize configurable in Phase 2.

File: src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    private ImageReader createImageReader(ImageInputStream is) throws IOException {
+        if (is == null || is.length() == 0) {
+            throw new IOException("Invalid input stream");
+        }
+
+        Iterator<ImageReader> iterator = ImageIO.getImageReaders(is);
+        if (iterator == null || !iterator.hasNext()) {
+            throw new IOException("Image file format not supported by ImageIO: ");
```
Comment (ljgeorge) [inline src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java:127]: Is the file image format being unsupported the only way the iterator might be empty? Would "No file readers found for image content" be a better wording?

File: src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                    LOG.debug("Confidence is " + confidence + " for blank line" );
+                }
+            } while (api.TessResultIteratorNext(ri, level) == ITessAPI.TRUE);
+        }
+        return pageText;
+    }
+
+    private ImageReader createImageReader(ImageInputStream is) throws IOException {
+        if (is == null || is.length() == 0) {
+            throw new IOException("Invalid input stream");
```
Comment (ljgeorge) [inline src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java:122]: Would "Empty image input stream" be a more helpful message?

**Review by ljgeorge — CHANGES_REQUESTED** [review]
The logic of the code seems fine. Probably worth running the SonarLint plugin over the code.

File: src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java (hunk 3 above)
Comment (SteveHicksCH) [reply src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java:122]: Agreed

File: src/main/java/uk/gov/companieshouse/ocr/api/ThreadConfig.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Configuration
+@EnableAsync
+public class ThreadConfig {
+
+    private final static int DEFAULT_THREAD_POOL_SIZE = 4;
```
[the remark this answers is not in the document]
Comment (SteveHicksCH) [reply src/main/java/uk/gov/companieshouse/ocr/api/ThreadConfig.java:15]: Done

File: src/main/java/uk/gov/companieshouse/ocr/api/ThreadConfig.java (hunk 1 above)
Comment (SteveHicksCH) [reply src/main/java/uk/gov/companieshouse/ocr/api/ThreadConfig.java:37]: As per discussion it is now using Environmental Reader

File: src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java (hunk 2 above)
Comment (SteveHicksCH) [reply src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java:127]: Updated

**Comment by SteveHicksCH** [comment]
Code updated and moved on

**Pull request closed without merging**